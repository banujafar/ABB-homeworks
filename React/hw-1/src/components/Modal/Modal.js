import React, { useState } from "react";
import "./Modal.scss";
class Modal extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="transparent" onClick={this.props.onClose}>
        <div className="modal">
          <div className="modal-top">
            <span className="close-button">x</span>
            <div className="header">{this.props.header}</div>
          </div>
          <div className="modal-bottom">
            <div className="text">{this.props.text}</div>
            <div className="actions" onClick={this.props.onClose}>
              {this.props.actions}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Modal;
