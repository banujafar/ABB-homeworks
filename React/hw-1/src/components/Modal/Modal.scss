.transparent {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
}

.modal {
  border-radius: 4px;
  width: 516px;
  color: #fff;
  margin: 0 auto;

  .modal-top {
    font-size: 22px;
    display: flex;
    padding: 30px;
    justify-content: space-between;
    flex-direction: row-reverse;
    line-height: 60px;
    font-family: Arial, Helvetica, sans-serif;
    background-color: #d44637;

    .close-button {
      cursor: pointer;
    }

  }

  .modal-bottom {
    background-color: #e74c3c;
    padding: 30px;

    .text {
      font-size: 15px;
      line-height: 30px;
      text-align: center;
    }

    .actions {
      display: flex;
      margin: 40px auto 0;
      justify-content: center;
      align-items: center;
      
      .action {
        margin-right: 9px;
        padding: 15px 40px;
        border-radius: 4px;
        background-color: #b3382c;
        cursor: pointer;
      }
    }
  }
}
