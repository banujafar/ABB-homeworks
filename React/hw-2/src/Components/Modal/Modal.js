import React, { useState } from "react";
import PropTypes from "prop-types";
import styles from "./Modal.module.scss";
import Button from "../Button/Button";

class Modal extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      this.props.closeButton && (
        <div className={styles["transparent"]} onClick={this.props.onClose}>
          <div className={styles["modal"]}>
            <div className={styles["modal-top"]}>
              <span
                className={styles["close-button"]}
                onClick={this.props.onClose}
              >
                x
              </span>
              <div className={styles["header"]}>{this.props.header}</div>
            </div>

            <div className={styles["modal-bottom"]}>
              <div className={styles["text"]}>{this.props.text}</div>
              <div className={styles["actions"]}>{this.props.actions}</div>
            </div>
          </div>
        </div>
      )
    );
  }
}
Modal.propTypes = {
  header: PropTypes.string.isRequired,
  closeButton: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};
export default Modal;
