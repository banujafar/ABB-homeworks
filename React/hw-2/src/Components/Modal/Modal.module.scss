.transparent {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
}
.modal {
  border-radius: 4px;
  width: 510px;
  color: #000;
  margin: 0 auto;

  .modal-top {
    font-size: 22px;
    display: flex;
    padding: 10px 15px;
    justify-content: space-between;
    flex-direction: row-reverse;
    line-height: 60px;
    font-family: Arial, Helvetica, sans-serif;
    background-color: #fff;

    .close-button {
      cursor: pointer;
    }
  }

  .modal-bottom {
    background-color: #fff;
    padding: 10px;

    .text {
      font-size: 15px;
      line-height: 30px;
      text-align: center;
    }

    .actions {
      display: flex;
      margin: 40px auto 0;
      justify-content: center;
      align-items: center;
    }
  }
}
