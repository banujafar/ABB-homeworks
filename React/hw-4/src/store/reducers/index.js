import { combineReducers } from "redux";
import { modalReducer } from "./modalReducer";
import { productReducer } from "./productReducer";
import { cartReducer } from "./cartReducer";
import { favoritesReducer } from "./favoritesReducer";

export const rootReducer=combineReducers({
    modal:modalReducer,
    products:productReducer,
    cart:cartReducer,
    favorites:favoritesReducer
})