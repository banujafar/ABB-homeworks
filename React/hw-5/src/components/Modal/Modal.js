import React from "react";
import PropTypes from "prop-types";
import styles from "./Modal.module.scss";

const Modal = (props) => {
  return (
    <>
      <div className={styles["modal"]}>
        <div className={styles["modal-top"]}>
          <span className={styles["close-button"]} onClick={props.onClose}>
            x
          </span>

          <div className={styles["header"]}>{props.header}</div>
        </div>

        <div className={styles["modal-bottom"]}>
          <div className={styles["text"]}>{props.text}</div>
          <div className={styles["actions"]}>{props.actions}</div>
        </div>
      </div>
      <div className={styles["transparent"]} onClick={props.onClose}></div>
    </>
  );
};

Modal.propTypes = {
  header: PropTypes.string.isRequired,
  closeButton: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
};

export default Modal;
