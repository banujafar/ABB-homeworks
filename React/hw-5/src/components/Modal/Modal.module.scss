.transparent {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 1;
}

.modal {
  border-radius: 4px;
  width: 510px;
  color: #000;
  margin: 0 auto;
  z-index: 1000;
  position: fixed;
  top: 0;
  left: 35%;

  .modal-top {
    font-size: 22px;
    display: flex;
    padding: 10px 15px;
    justify-content: space-between;
    flex-direction: row-reverse;
    line-height: 60px;
    font-family: Arial, Helvetica, sans-serif;
    background-color: #fff;

    .close-button {
      cursor: pointer;
    }
  }

  .modal-bottom {
    background-color: #fff;
    padding: 10px;

    .text {
      font-size: 15px;
      line-height: 30px;
      text-align: center;
    }

    .actions {
      display: flex;
      margin: 40px auto 0;
      justify-content: center;
      align-items: center;
    }
  }
}
