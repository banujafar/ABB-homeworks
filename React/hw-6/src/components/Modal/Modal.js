import React from "react";
import PropTypes from "prop-types";
import styles from "./Modal.module.scss";

const Modal = (props) => {
  return (
    <>
      <div className={styles["modal"]} data-testid="modal">
        <div className={styles["modal-top"]}>
          <span className={styles["close-button"]} onClick={props.onClose}>
            x
          </span>

          <div className={styles["header"]} data-testid="modal-header">
            {props.header}
          </div>
        </div>

        <div className={styles["modal-bottom"]}>
          <div className={styles["text"]} data-testid="modal-text">
            {props.text}
          </div>
          <div className={styles["actions"]}>{props.actions}</div>
        </div>
      </div>
      <div className={styles["transparent"]} onClick={props.onClose}></div>
    </>
  );
};

Modal.propTypes = {
  header: PropTypes.string,
  closeButton: PropTypes.bool,
  text: PropTypes.string,
};

export default Modal;
