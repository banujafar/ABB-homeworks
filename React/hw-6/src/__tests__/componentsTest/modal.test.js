import { fireEvent, waitFor, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store";
import Modal from "../../components/Modal/Modal";

test("should render modal component", () => {
  const propsHeader = "modal-header";
  const onCloseMock = jest.fn();
  render(
    <Provider store={store}>
      <Modal header={propsHeader} onClose={onCloseMock} />
    </Provider>
  );
  const modal = screen.getByTestId("modal");
  expect(modal).toBeInTheDocument();
  const modalHeader = screen.getByTestId("modal-header");
  expect(modalHeader.textContent).toBe(propsHeader);
  const closeBtn = screen.getByText("x");
  expect(closeBtn).toBeInTheDocument();
  fireEvent.click(closeBtn);
  fireEvent.click(closeBtn);
  expect(onCloseMock).toHaveBeenCalledTimes(2);
});

test("should matches DOM snapshot correctly", () => {
  const propsHeader = "modal-header";
  const onCloseMock = jest.fn();
  const tree = render(<Modal header={propsHeader} onClose={onCloseMock} />);
  expect(tree).toMatchSnapshot();
});
