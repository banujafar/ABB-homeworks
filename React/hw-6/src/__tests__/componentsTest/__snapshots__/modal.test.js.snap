// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches DOM snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal"
        data-testid="modal"
      >
        <div
          class="modal-top"
        >
          <span
            class="close-button"
          >
            x
          </span>
          <div
            class="header"
            data-testid="modal-header"
          >
            modal-header
          </div>
        </div>
        <div
          class="modal-bottom"
        >
          <div
            class="text"
            data-testid="modal-text"
          />
          <div
            class="actions"
          />
        </div>
      </div>
      <div
        class="transparent"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="modal"
      data-testid="modal"
    >
      <div
        class="modal-top"
      >
        <span
          class="close-button"
        >
          x
        </span>
        <div
          class="header"
          data-testid="modal-header"
        >
          modal-header
        </div>
      </div>
      <div
        class="modal-bottom"
      >
        <div
          class="text"
          data-testid="modal-text"
        />
        <div
          class="actions"
        />
      </div>
    </div>
    <div
      class="transparent"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
